providers/ldap/ldap_client.go

```diff
@@ -0,0 +1,17 @@
+package ldap
+
+import (
+	"net/http"
+
+	"github.com/rancher/rancher-auth-service/model"
+)
+
+// LClient implements a httpclient for github
+type LClient struct {
+	httpClient *http.Client
+	config     *model.LdapConfig
```

Comment:
Actually looking at ldap_provider.go, I feel all the methods in there are using Lclient.config. All those methods accept the l *LdapProvider and then use the l.ldapClient.config from the passed object, nothing else is needed from *LdapProvider. So seems like those methods should be defined on this struct.